bpf/bpf_overlay.c

```diff
@@ -150,6 +150,36 @@ static __always_inline int handle_ipv6(struct __ctx_buff *ctx,
 not_esp:
 #endif
 
+#if defined(ENABLE_EGRESS_GATEWAY_COMMON)
+	{
+		__u32 egress_ifindex = 0;
+		union v6addr snat_addr, daddr;
+
+		if (egress_gw_snat_needed_hook_v6((union v6addr *)&ip6->saddr,
+						  (union v6addr *)&ip6->daddr, &snat_addr,
+						  &egress_ifindex)) {
```

Comment:
```suggestion
                ipv6_addr_copy(&daddr, ip6->daddr);
		if (egress_gw_snat_needed_hook_v6((union v6addr *)&ip6->saddr,
						  &daddr, &snat_addr,
						  &egress_ifindex)) {
```

This is probably also needed to make the FIB lookup work as expected (where we pass `daddr`).